src/private/privateAPIs.ts

```diff
@@ -203,3 +204,18 @@ export function getConfigSetting(callback: (value: string) => void, key: string)
 
   sendMessageToParent('getConfigSetting', [key], callback);
 }
+
+/**
+ * @private
+ * register a handler to be called when a user setting changes. The changed setting key & value is provided in the callback.
+ * @param settingKeys List of user setting changes to subscribe
+ * @param handler When a subscribed setting is updated this handler is called
+ */
+export function registerUserSettingsChangeHandler(
+  settingKeys: UserSettingKeys[],
+  handler: (settingKey: UserSettingKeys, value: any) => void,
```

Comment:
with this design we are expecting handler to execute for each setting change. Which make sense.